### PR TITLE
Submits NC Games Ticket 7 for review

### PR DESCRIPTION
**Pull request by stef-lukes**
# 7. GET /api/reviews/:review_id (comment count)

- amended app.test.js : edited previous test for same endpoint from strict object equality to match object instead.  Created new test for same endpoint to match object with new comment count key value.
- amended reviews.model.js db query to now join relevant data from comments to review data request.
- tests passing

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -47,14 +47,13 @@ describe("/api", () => {
...
-  describe("4 GET/api/reviews/:reviewId", () => {
+  describe("4 GET/api/reviews/:review_id", () => {
     test("status 200, should respond with an object containing the properties: review_id, title, review_body, designer, review_img_url, votes, category, owner, created_at", () => {
-      const REVIEW_ID = 1;
       return request(app)
         .get(`/api/reviews/1`)
         .expect(200)
         .then(({ body }) => {
-          expect(body.review).toEqual({
+          expect(body.review).toMatchObject({
```
Comment (mallionaire) [inline __tests__/app.test.js:56]: Please refactor this to .toEqual and check that all the properties are on the object, this isn't really secure enough to know that you're not receiving anything else by mistake

File: __tests__/app.test.js [hunk 2]
```diff
@@ -192,3 +191,34 @@ describe("/api", () => {
...
+
+// ====================================
+
+describe("7. GET /api/reviews/:review_id (comment count)", () => {
+  test("status 200, should now respond with the same object as in TEST BLOCK 4 with the addition of a COMMENTS_COUNT property, which counts the number references to the review_id passed in the comments data", () => {
+    return request(app)
+      .get(`/api/reviews/3`)
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.review).toMatchObject({
```
Comment (mallionaire) [inline __tests__/app.test.js:203]: see prev comment about .toMatchObj

File: models/reviews.model.js [hunk 3]
```diff
@@ -2,15 +2,23 @@ const db = require("../db/connection");
...
+    .query(
+      `SELECT reviews.*, COUNT(comments.review_id):: INT AS comment_count 
+    FROM reviews 
+    LEFT JOIN comments ON reviews.review_id = comments.review_id
+    WHERE reviews.review_id = $1
+    GROUP BY reviews.review_id`,
+      [review_id]
+    )
+    .then((review) => {
+      console.log(review.rows);
```
Comment (mallionaire) [inline models/reviews.model.js:14]: remove console.logs

**Review by mallionaire — CHANGES_REQUESTED** [review]

File: __tests__/app.test.js (hunk 1 above)
Comment (stef-lukes) [reply __tests__/app.test.js:56]: So both test block 4 and test block 7 should expect strict equality to all the key values?

**Review by Rosem10 — APPROVED** [review]

**Pull request merged**